File: MvpFramework/src/main/java/com/ytx/mvpframework/presenter/FragmentPresenter.java
package com.ytx.mvpframework.presenter;

import android.arch.lifecycle.LifecycleOwner;
import android.support.v4.app.Fragment;

import com.ytx.mvpframework.view.IView;

public class FragmentPresenter<V extends IView> extends LifecyclePresenter<V> {
    private boolean isViewCreated = false;

    public FragmentPresenter(V view) {
        super(view);
        if (view instanceof Fragment) {
            bindFragment((Fragment) view);
        }
    }

    public final void bindFragment(Fragment view) {
        bindLifecycle(view);
    }

    public final void unbindFragment() {
        unbindLifecycle();
    }

    public void onViewCreated(LifecycleOwner owner) {
        isViewCreated = true;
    }

    public void onDestroyView(LifecycleOwner owner) {
        isViewCreated = false;
    }

    public final boolean isViewCreated() {
        return isViewCreated;
    }
}

File: MvpFramework/src/main/java/com/ytx/mvpframework/presenter/FragmentPresenter.kt
//@file:JvmName("FragmentPresenter")
//package com.ytx.mvpframework.presenter
//
//import android.arch.lifecycle.LifecycleOwner
//import android.support.v4.app.Fragment
//import com.ytx.mvpframework.view.IView
//
//open class FragmentPresenter<V : IView>(view: V) : LifecyclePresenter<V>(view) {
//    private var isViewCreated: Boolean = false
//
//    init {
//        if (view is Fragment) {
//            bindFragment(view)
//        }
//    }
//
//    public fun bindFragment(view: Fragment) {
//        bindLifecycle(view)
//    }
//
//    public fun unbindFragment() {
//        unbindLifecycle()
//    }
//
//    open fun onViewCreated(owner: LifecycleOwner) {
//        isViewCreated = true
//    }
//
//    open fun onDestroyView(owner: LifecycleOwner) {
//        isViewCreated = false
//    }
//
//    fun isViewCreated() : Boolean {
//        return isViewCreated
//    }
//}